frontend/public/components/storage/attach-storage.tsx

```diff
@@ -40,306 +27,306 @@ const PVCDropdown: React.SFC<PVCDropdownProps> = props => {
 
-  onMountAsReadOnlyChanged: React.ReactEventHandler<HTMLInputElement> = () => {
-    this.setState({ mountAsReadOnly: !this.state.mountAsReadOnly });
+  const onMountAsReadOnlyChanged: React.ReactEventHandler<HTMLInputElement> = () => {
+    setMountAsReadOnly(!mountAsReadOnly);
   };
 
-  createPVCIfNecessary(): Promise<string> {
-    const { showCreatePVC, newPVCObj, claimName } = this.state;
+  const createPVCIfNecessary = () => {
     return showCreatePVC === 'new'
       ? k8sCreate(PersistentVolumeClaimModel, newPVCObj).then(claim => claim.metadata.name)
       : Promise.resolve(claimName);
-  }
+  };
 
-  save = (event: React.FormEvent<EventTarget>) => {
-    event.preventDefault();
-    const { kindObj } = this.props;
-    const {
-      resourceObj: originalObj,
-      volumeName,
+  const getVolumePatches = (pvClaimName: string) => {
+    const mount = {
+      name: volumeName,
       mountPath,
       subPath,
-      mountAsReadOnly,
-      volumeAlreadyMounted,
-    } = this.state;
-    const { metadata } = originalObj;
-    const resourceObj = _.cloneDeep(originalObj);
-    const template = resourceObj.spec.template;
-
-    // For each container in the pod spec, add the new volume mount.
-    _.each(template.spec.containers, container => {
-      if (!this.isContainerSelected(container)) {
-        return;
-      }
-      if (!container.volumeMounts) {
-        container.volumeMounts = [];
-      }
-      container.volumeMounts.push({
-        name: volumeName,
-        mountPath,
-        subPath,
-        readOnly: mountAsReadOnly,
-      });
-    });
+      readOnly: mountAsReadOnly,
+    };
 
-    this.setState({ inProgress: true });
-    this.createPVCIfNecessary().then(claimName => {
-      // add new volume to the pod template if not preent
-      if (!volumeAlreadyMounted) {
-        template.spec.volumes = template.spec.volumes || [];
-        template.spec.volumes.push({
-          name: volumeName,
-          persistentVolumeClaim: {
-            claimName,
-          },
-        });
+    const containers: ContainerSpec[] = _.get(obj, 'spec.template.spec.containers', []);
+    const patches = containers.map((container, i) => {
+      // Only add to selected containers
+      if (isContainerSelected(container)) {
+        return _.isEmpty(container.volumeMounts)
+          ? {op: 'add', path: `/spec/template/spec/containers/${i}/volumeMounts`, value: [mount]}
+          : {op: 'add', path: `/spec/template/spec/containers/${i}/volumeMounts/-`, value: mount};
       }
-      k8sUpdate(kindObj, resourceObj, metadata.namespace, metadata.name).then(
-        resource => {
-          this.setState({ inProgress: false });
-          history.push(resourceObjPath(resource, referenceFor(resource)));
-        },
-        err => this.setState({ error: err.message, inProgress: false })
-      );
-    }, err => this.setState({ error: err.message, inProgress: false }));
+    });
+    const filteredPatches = patches.filter(Boolean);
```

Comment:
I get undefined entries in my patches array when not all containers are selected.  I needed to get rid of them for the `k8sPatch` to work so I used the filter(Boolean) command.  I am open to sugestions of how to not get those undefined values in there to begin with.

Reply:
@zherman0 use `reduce` instead of `map`, then you don't have to include all containers in the patches. Something like this (untested)

```js
   const patches = containers.reduce((patches, container, i) => {
      // Only add to selected containers
      if (isContainerSelected(container)) {
	if (_.isEmpty(container.volumeMounts) {
	  patches.push({op: 'add', path: `/spec/template/spec/containers/${i}/volumeMounts`, value: [mount]});
	} else {
	  patches.push({op: 'add', path: `/spec/template/spec/containers/${i}/volumeMounts/-`, value: mount});
	}
      }
	  
      return patches;
    }, []);
```